### PR TITLE
ssd1306: add driver with optimizations

### PR DESCRIPTION
**Pull request by joy4eg**
Summary:
-  Add base ssd1306 driver from PR #37 
- Optimizations:
  - Remove unnecessary global variables
  - Fix memory leak
  - Add deferred io support
  - Fix code style

### CODE REVIEW

File: ssd1306/ssd1306_main.c [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+}
+
+static void ssd1306_display_update_work(struct work_struct *work)
+{
+	struct ssd1306_ctx *ctx = container_of(work, struct ssd1306_ctx, display_ws);
+
+	ssd1306_display_update_direct(ctx);
+}
+
+static void ssd1306_display_update(struct ssd1306_ctx *ctx)
```
Comment (DevyatovAndrey) [inline ssd1306/ssd1306_main.c]: There are several *_display_update_* methods - that's really confusing. Could you rename these?

File: ssd1306/ssd1306_main.c [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+
+	if (count + p > total_size)
+		count = total_size - p;
+
+	if (!count)
+		return -EINVAL;
+
+	dst = (void __force *) (info->screen_base + p);
+
+	if (copy_from_user(dst, buf, count))
```
Comment (DevyatovAndrey) [inline ssd1306/ssd1306_main.c]: I think there could be used a *fb_sys_write* method

**Review by DevyatovAndrey — CHANGES_REQUESTED** [review]
Good work in general!
But still it could you please make changes listed below?

**Comment by joy4eg** [comment]
Fixed.

**Review by AleksandrBulyshchenko — CHANGES_REQUESTED** [review]
@joy4eg,
- Please base your changes onto original #37 branch (or its merge) instead of copying final version;
- Split the PR into functional commits with descriptive commit messages;

Format your commits according to the [wiki](https://github.com/Kernel-GL-HRK/gl-kernel-training-2017/wiki/Pull-Request-Guideline).

**Pull request closed without merging**